### ISSUE TITLE
UnicodeDecodeError

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools-scm#26]
I have a python project. Trying to execute
`py2 setup.py bdist_wheel`

```
<skipping lines...>
  File "setup.py", line 20, in <module>
    description='my script'
  File "C:\Python27\ArcGIS10.2\lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "C:\Python27\ArcGIS10.2\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "C:\Python27\ArcGIS10.2\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\wheel\bdist_wheel.py", line 211, in run
    self.run_command('install')
  File "C:\Python27\ArcGIS10.2\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\Python27\ArcGIS10.2\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\install.py", line 61, in run
    return orig.install.run(self)
  File "C:\Python27\ArcGIS10.2\lib\distutils\command\install.py", line 575, in run
    self.run_command(cmd_name)
  File "C:\Python27\ArcGIS10.2\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\Python27\ArcGIS10.2\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\install_egg_info.py", line 33, in run
    self.run_command('egg_info')
  File "C:\Python27\ArcGIS10.2\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\Python27\ArcGIS10.2\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\egg_info.py", line 180, in run
    self.find_sources()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\egg_info.py", line 207, in find_sources
    mm.run()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\egg_info.py", line 291, in run
    self.add_defaults()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\egg_info.py", line 323, in add_defaults
    rcfiles = list(walk_revctrl())
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools\command\sdist.py", line 19, in walk_revctrl
    for item in ep.load()(dirname):
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools_scm\__init__.py", line 77, in find_files
    return do(ep.load(), path).splitlines()
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools_scm\utils.py", line 48, in do
    out, err, ret = do_ex(cmd, cwd)
  File "C:\Python27\ArcGIS10.2\lib\site-packages\setuptools_scm\utils.py", line 44, in do_ex
    return out.strip().decode(), err.strip().decode(), p.returncode
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf1 in position 17: ordinal not in range(128)
```

Windows 7, python 2.7

---
- Bitbucket: https://bitbucket.org/pypa/setuptools_scm/issue/26
- Originally reported by: [lansman](http://bitbucket.org/lansman)
- Originally created at: 2015-05-19T07:56:32.400

**Comment by jaraco** [comment]
It is bad practice to use string' decode() without explicit setting encoding, because on some platforms (including Windows) decode' encoding is ascii, so amount of supporting characters is VERY tiny.

You should consider to change all `.decode()` to `.decode(locale.getpreferredencoding())` or `.decode(sys.stdin.encoding)` or `.decode(sys.stdout.encoding)` depending on where your string come from.

---

Original comment by: [lansman](http://bitbucket.org/lansman)

**Comment by jaraco** [comment]
Propper decode handling, fixes #26

→ <<cset c38975206ccd>>

---

Original comment by: Ronny Pfannschmidt
